src/hooks/xcm/useXcmBridge.ts

```diff
@@ -333,6 +339,56 @@ export function useXcmBridge(selectedToken?: Ref<ChainAsset>) {
     }
   };
 
+  const transferAsset = async (transferAmt: number, toAddress: string): Promise<void> => {
+    try {
+      if (!selectedToken?.value) {
+        throw Error('Token is not selected');
+      }
+
+      const isValidSS58Address =
+        isValidAddressPolkadotAddress(currentAccount.value) &&
+        isValidAddressPolkadotAddress(toAddress);
+
+      if (!isValidSS58Address && !isValidEvmAddress(toAddress)) {
+        store.dispatch('general/showAlertMsg', {
+          msg: 'assets.invalidAddress',
+          alertType: 'error',
+        });
+        return;
+      }
+
+      const receivingAddress = isValidEvmAddress(toAddress) ? toSS58Address(toAddress) : toAddress;
+
+      const txResHandler = async (result: ISubmittableResult): Promise<boolean> => {
+        const res = await handleResult(result);
+        await handleUpdateTokenBalances();
+        return res;
+      };
+
+      const decimals = selectedToken.value.metadata.decimals;
+      const transaction = $api!.tx.assets.transfer(
+        new BN(selectedToken.value.id),
+        receivingAddress,
+        new BN(10 ** Number(decimals)).muln(transferAmt)
```

Comment:
From now on, let's [use ethers.utils](https://docs.ethers.io/v5/api/utils/display-logic/) for decimals calculation [like this](https://github.com/AstarNetwork/astar-apps/blob/4bde37846fd0b48f6b774c4a31360f4c65088438/src/hooks/xcm/useXcmBridge.ts#L320) to avoid missing calculation. I'll include this in the next PR. :)

Ref:
 * https://github.com/AstarNetwork/astar-apps/pull/244#discussion_r797643314

![image](https://user-images.githubusercontent.com/92044428/168953056-2d990712-2e4c-4d6f-b04b-482d9c8fdf52.png)

CC: @sirius651 @hoonsubin